src/main/java/matching/MatchingAlgorithm.java

```diff
@@ -13,63 +14,109 @@ public class MatchingAlgorithm {
 
         // assign scores to each user
-        for (int i = 0; i < otherUsers.size(); i++){
+        for (int i = 0; i < otherUsers.size(); i++) {
             Account oUser = otherUsers.get(i);
-            for (String key: preferences.keySet()){
-                double score = oUser.getProfile().assignScore(key, preferences, oUser.getProfile());
-
+            for (String key : preferences.keySet()) {
+                assignScore(key, preferences, oUser.getProfile());
             }
         }
 
         // sort users based on descending order of mathscore
-        matches = new List<users>(); //potential matches
-        Collections.sort(otherUsers, new matchScoreComparator());
+//        ArrayList<Account> matches = new ArrayList<Account>(); //potential matches
+        Collections.sort(otherUsers);
+
 
         // create new LinkedList
-        LinkedList<Account> ret = new LinkedList<Account>();
+        LinkedList<Account> potentialMatches = new LinkedList<>(otherUsers);
 
-        return matches; //potential matches
+        return potentialMatches;
     }
 
     //should make new class (class file?) for both of the below? or just make comparator a static method?
-
+//    @Override
+//    public int compareTo(Account other){
+//        return profile.getScore().compareTo(other.getProfile().getScore());
+//
+//    }
 
     // sorter
-    public static void sort(ArrayList<Account> list) {
-
-        list.sort((o1, o2) -> o1.getProfile.getScore().compareTo(
-                o2.getProfile.getScore()));
-    }
+//    public static void sort(ArrayList<Account> list) {
+//
+//        list.sort((o1, o2) -> compare(o1, o2));
+//    }
+//
+//    // example: https://mkyong.com/java/java-object-sorting-example-comparable-and-comparator/#:~:text=To%20sort%20an%20Object%20by,the%20new%20Fruit%20class%20again.&text=The%20new%20Fruit%20class%20implemented,quantity%20property%20in%20ascending%20order.
+//    // override compare users by comparing their scores
+//    @Override
+//    public static class matchScoreComparator implements Comparator<Account> {
+//
+//
+//        public int compare(Account a1, Account a2) {
+//            return a1.getProfile().getScore().compareTo(a2.getProfile().getScore());
+//        }
+//    }
+
+    public static void assignScore(String key, HashMap preferences, Profile oUser) {
+
+
+        // can change this to apply for all keys, need to see how joy implements preferences
+        // if they are all sorted then can apply and dont need to check if key =
+
+        // 3 keys: year, field, style
+        // year preferences are sorted
+        int score = oUser.getScore();
+
+        //compare years
+        if (Objects.equals(key, "year")) {
+            ArrayList<String> years = (ArrayList<String>) preferences.get("year");
+            for (int i = 0; i < years.size(); i++) {
+                if (years.get(i) == oUser.getYear()) {
```

Comment:
might have to use the equals() method since you're trying to compare the contents of the two Strings, not the mem addresses